app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -174,9 +186,9 @@ object RegisterController {
 
   object EligibleInfo {
 
-    case class EligibleWithEmail(userInfo: UserInfo, email: Email, confirmedEmail: Option[Email], eligible: Eligible) extends EligibleInfo
+    case class EligibleWithEmail(userInfo: UserInfo, confirmedEmail: Option[Email], eligible: Eligible, bankDetails: Option[BankDetails]) extends EligibleInfo
 
-    case class EligibleWithNoEmail(userInfo: UserInfo, eligible: Eligible) extends EligibleInfo
+    case class EligibleWithNoEmail(userInfo: UserInfo, eligible: Eligible, bankDetails: Option[BankDetails]) extends EligibleInfo
```

Comment:
would we not need bank details here as well? The email here was supposed to be the email in `userInfo` if it was defined